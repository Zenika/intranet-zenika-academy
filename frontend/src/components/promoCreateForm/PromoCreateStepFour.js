import React, { Component } from 'react';
import CsvPicker from '../csvPicker/CsvPicker';
import './PromoCreate.scss';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';

export class PromoCreateStepFour extends Component {
  render() {
    const {
      nextStep,
      prevStep,
      step,
      handleCSVImport,
      name,
      promo,
      edit,
    } = this.props;
    const selected = !!(promo.students.length && promo.students.length > 0);

    const buttonForm = (
      <section className="field buttonField section">
        <section className="control">
          <button
            id="previousButton"
            type="button"
            className="button is-danger"
            onClick={prevStep}
          >
            Revenir
          </button>
        </section>
        <section className="control">
          <button
            id="resumeButton"
            type="button"
            className="button is-link"
            onClick={nextStep}
          >
            Continuer
          </button>
        </section>
      </section>
    );

    return (
      <div className="promoCreateForm">
        <article className="section box">
          <h1 className="title is-2 is-spaced">{`${
            edit ? 'Edition' : 'Création'
          } d'une promo`}</h1>
          <BulmaSteps step={step} />
          <section className="control">
            <span htmlFor="students" className="label">
              Importer des étudiants :
            </span>
            <div className="csvPickerContainer">
              <CsvPicker
                selected={selected}
                name={name}
                handleCSVImport={handleCSVImport}
              />
            </div>
          </section>
          {buttonForm}
        </article>
      </div>
    );
  }
}

export default PromoCreateStepFour;
