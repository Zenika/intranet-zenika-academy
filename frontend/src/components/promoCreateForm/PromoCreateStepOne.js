import React from 'react';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import DatePicker from '../datepicker/DatePicker';
import './PromoCreate.scss';

function PromoCreateStepOne(props) {
  const { nextStep, step, handleChange, promo, edit } = props;

  const buttonForm = (
    <div className="field buttonField">
      <div className="control">
        <a
          href="/home/admin/"
          id="cancelButton"
          type="button"
          className="button is-danger"
        >
          Annuler
        </a>
      </div>
      <div className="control">
        <button
          id="resumeButton"
          type="button"
          className="button is-link"
          onClick={nextStep}
        >
          Continuer
        </button>
      </div>
    </div>
  );

  return (
    <div className="promoCreateForm">
      <article className="div box">
        <h1 className="title is-2 is-spaced">{`${
          edit ? 'Edition' : 'Création'
        } d'une promo`}</h1>
        <BulmaSteps step={step} />
        <div className="control">
          <label htmlFor="promoName" className="label">
            Nom de la promo:
            <input
              id="promoName"
              className="input "
              name="title"
              defaultValue={promo.title}
              onChange={(e) => handleChange(e)}
              type="text"
              placeholder="Promo"
            />
          </label>
        </div>
        <div className="level">
          <div className="control">
            <span className="label">Date de début: </span>
            <DatePicker
              date={promo.startDate}
              name="startDate"
              defaultValue={promo.startDate}
              handleChange={(e) => handleChange(e)}
            />
          </div>
          <div className="control">
            <span className="label">Date de fin: </span>
            <DatePicker
              date={promo.endDate}
              name="endDate"
              defaultValue={promo.endDate}
              handleChange={(e) => handleChange(e)}
            />
          </div>
        </div>
        <div className="field">
          <label htmlFor="city" className="label">
            Ville:
            <div className="field">
              <input
                type="text"
                id="city"
                className="input"
                defaultValue={promo.city}
                name="city"
                onChange={(e) => handleChange(e)}
                placeholder="Ville"
              />
            </div>
          </label>
        </div>
        {buttonForm}
      </article>
    </div>
  );
}

export default PromoCreateStepOne;
