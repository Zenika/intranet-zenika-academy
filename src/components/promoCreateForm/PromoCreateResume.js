import React, { Component } from 'react'
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import './PromoCreate.scss';


class PromoCreateResume extends Component {
  render() {
    const { step, promo, prevStep } = this.props;

    console.log(promo);


    const startDate = promo.startDate.split("-").reverse().join("-");
    const endDate = promo.endDate.split("-").reverse().join("-");


    const buttonForm = (
      <section className="field buttonField">
        <section className="control">
          <button className="button is-danger" onClick={prevStep}>Revenir</button>
        </section>
        <section className="control">
          <button className="button is-link" >Valider</button>
        </section>
      </section>
    );

    return (
      <div className="promoCreateForm">
        <article className="section box">
          <h1 className="title is-4 is-spaced">Création d'une promo</h1>
          <BulmaSteps step={step} />
          <section className="field">
            <span className="title is-4 is-spaced">Résumé</span>
          </section>
          <section className="control">
            <section className="field">
              <label className="label">Nom de la promo: </label>
              <section className="field">
                <p>{promo.title ? promo.title : ""}</p>
              </section>
            </section>
          </section>
          <section>
            <section className="control">
              <section className="field">
                <label className="label">Date de début: </label>
                <section className="field">
                  <span> <time >{startDate ? startDate : ""}</time> </span>
                </section>
              </section>
            </section>
          </section>
          <section>
            <section className="control">
              <section className="field">
                <label className="label">Date de fin: </label>
                <section className="field">
                  <span> <time >{endDate ? endDate : ""}</time> </span>
                </section>
              </section>
            </section>
          </section>
          <section className="field">
            <label className="label">Pays:</label>
            <section className="field">
              <p>{promo.country.label !== undefined ? promo.country.label : ""}</p>
            </section>
          </section>
          <section className="field">
            <label className="label">Ville:</label>
            <section className="field">
              <p>{promo.city.label !== undefined ? promo.city.label : ""}</p>
            </section>
          </section>
          <section className="field">
            <label className="label">Programme:</label>
            <section className="field">
              <p>{promo.program.label !== undefined ? promo.program.label : ""}</p>
            </section>
          </section>
          <section className="field">
            <label className="label">Formateurs:</label>
            <section className="field">
              {promo.teachers.length ? promo.teachers.map((e, i) => {
                return <p key={i}>{e.Nom ? `${e.Nom} ${e.Prénom}` : e.label}</p>
              }) : ""}
            </section>
          </section>
          <section className="field">
            <label className="label">Elèves:</label>
            <section className="field">
              {promo.students.length ? promo.students.map(e => {
                return <p key={e.Nom}>{`${e.Nom}  ${e.Prénom}  ${e.Email}`}</p>
              }) : ""}
            </section>
          </section>
          {buttonForm}
        </article>
      </div>
    )
  }
}

export default PromoCreateResume;
