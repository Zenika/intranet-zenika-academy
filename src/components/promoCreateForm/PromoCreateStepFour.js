import React, { Component } from 'react';
import './PromoCreate.scss';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import SearchbarAutoComplete from '../searchbarauto/SearchbarAuto';

const students = [
  {
    label: 'Jeremy Pluquet',
    value: 'Jeremy Pluquet',
  },
  {
    label: 'Youcef Messaoudene',
    value: 'Youcef Messaoudene',

  },
  {
    label: 'Anne-Lise Vanhoegaerden',
    value: 'Monsieur X',
  },
  {
    label: 'Claudine Lafine',
    value: 'Claudine Lafine',
  },
];

export class PromoCreateStepFour extends Component {
  render() {
    const {
      nextStep, prevStep, step,
    } = this.props;

    const buttonForm = (
      <section className="field buttonField">
        <section className="control">
          <button className="button is-danger" onClick={prevStep}>Revenir</button>
        </section>
        <section className="control">
          <button className="button is-link" onClick={nextStep}>Continuer</button>
        </section>
      </section>
    );

    return (
      <div className="promoCreateForm">
        <article className="section box">
          <h1 className="title is-4 is-spaced">Création d'une promo</h1>
          <BulmaSteps step={step} />
          <section className="control">
            <label className="label">
              Choisir des élèves existants:
                        </label>
            <section className="field">
              <section className="control">
                <SearchbarAutoComplete searchObject={students} searchKey="name" isMulti />
              </section>
            </section>
          </section>
          <section className="field">
            <label className="label middleLines"><span>OU</span></label>
          </section>
          <section className="field">
            <section className="field">
              <section className="control buttonCreate">
                <button className="button is-link">Créer un utilisateur</button>
              </section>
            </section>
          </section>
          {buttonForm}
        </article>
      </div>
    );
  }
}

export default PromoCreateStepFour;
