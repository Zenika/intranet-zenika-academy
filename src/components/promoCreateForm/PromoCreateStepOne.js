import React from 'react';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import DatePicker from '../datepicker/DatePicker';
import './PromoCreate.scss';
import SearchbarAutoComplete from '../searchbarauto/SearchbarAuto';
import CreatableSelect from '../searchbarauto/CreatableSearchbar';

// const country = [
//   {
//     label: 'France',
//     value: 'France',
//   },
//   {
//     label: 'Maroc',
//     value: 'Maroc',

//   },
//   {
//     label: 'Espagne',
//     value: 'Espagne',
//   },
// ];

// const city = [
//   {
//     label: 'Paris',
//     value: 'France',
//   },
//   {
//     label: 'Rennes',
//     value: 'France',

//   },
//   {
//     label: 'Rabat',
//     value: 'Maroc',
//   },
// ];

function PromoCreateStepOne(props) {
  const {
    nextStep, step, handleChange, promo, handleMultiChange,
  } = props;

  const buttonForm = (
    <section className="field buttonField section">
      <section className="control">
        <button className="button is-danger">Annuler</button>
      </section>
      <section className="control">
        <button className="button is-link" onClick={nextStep}>Continuer</button>
      </section>
    </section>
  );

  return (
    <div className="promoCreateForm">
      <article className="section box">
        <h1 className="title is-4 is-spaced">Création d'une promo</h1>
        <BulmaSteps step={step} />
        <section className="control">
          <label className="label">Nom de la promo: </label>
          <input className="input " name="title" defaultValue={promo.title} onChange={(e) => handleChange(e)} type="text" placeholder="Promo" />
        </section>
        <section>
          <section className="control">
            <label className="label">Date de début: </label>
            <DatePicker date={promo.startDate} name="startDate" defaultValue={promo.startDate} handleChange={(e) => handleChange(e)} />
          </section>
        </section>
        <section>
          <section className="control">
            <label className="label">Date de fin: </label>
            <DatePicker date={promo.endDate} name="endDate" defaultValue={promo.endDate} handleChange={(e) => handleChange(e)} />
          </section>
        </section>
        {/* <section className="field">
          <label className="label">Pays:</label>
          <CreatableSelect defaultValue={promo.country} name="country" options={country} handleChange={(e) => handleMultiChange(e, 'country')} searchKey="country" defaultLabel="Pays" />
        </section> */}
        <section className="field">
          <label className="label">Ville:</label>
          <section className="field">
            <input className="input" defaultValue={promo.city} name="city" isLoading={false} onChange={(e) => handleChange(e)} defaultLabel="Ville" />
          </section>
        </section>
        {buttonForm}
      </article>
    </div>
  );
}


export default PromoCreateStepOne;
