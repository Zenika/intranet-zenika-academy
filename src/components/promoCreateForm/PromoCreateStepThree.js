import React, { Component } from 'react';
import './PromoCreate.scss';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import SearchbarAutoComplete from '../searchbarauto/SearchbarAuto';
const teachers = [
  {
    label: 'Jérémie Patonnier',
    value: 'Jérémie Patonnier',
  },
  {
    label: 'Jonathan Barthelemy',
    value: 'Jonathan Barthelemy',

  },
  {
    label: 'Monsieur X',
    value: 'Monsieur X',
  },
  {
    label: 'Madame Y',
    value: 'Madame Y',
  },
];


export class PromoCreateStepThree extends Component {
  render() {
    const {
      nextStep, prevStep, step, handleChange, promo, handleMultiChange,
    } = this.props;
    const buttonForm = (
      <section className="field buttonField section">
        <section className="control">
          <button className="button is-danger" onClick={prevStep}>Revenir</button>
        </section>
        <section className="control">
          <button className="button is-link" onClick={nextStep}>Continuer</button>
        </section>
      </section>
    );

    return (
      <div className="promoCreateForm">
        <article className="section box">
          <h1 className="title is-4 is-spaced">Création d'une promo</h1>
          <BulmaSteps step={step} />
          <section className="control">
            <label className="label">
              Choisir des formateurs existants:
            </label>
            <section className="field">
              <section className="control">
                <SearchbarAutoComplete defaultValue={promo.teachers} name="teachers" options={teachers} handleChange={(e) => handleMultiChange(e, 'teachers')} searchKey="title" defaultLabel="Formateurs" isMulti />
              </section>
            </section>
          </section>
          <div className="field section">
            <label className="label middleLines"><span>OU</span></label>
          </div>
          <section className="field">
            <section className="field">
              <section className="control buttonCreate">
                <button className="button is-link">Créer un utilisateur</button>
              </section>
            </section>
          </section>
          {buttonForm}
        </article>
      </div>
    );
  }
}

export default PromoCreateStepThree;
