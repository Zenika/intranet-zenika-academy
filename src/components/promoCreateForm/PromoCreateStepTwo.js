import React, { Component } from 'react';
import axios from 'axios';
import { BulmaSteps } from '../bulma-steps/BulmaSteps';
import CreatableSelect from '../searchbarauto/CreatableSearchbar';
import './PromoCreate.scss';

export class PromoCreateStepTwo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      programs: [],
    };
  }

  componentDidMount() {
    axios.get('http://localhost:4000/api/programs')
      .then((res) => res.data.forEach((program) => {
        if (program.type === 1) {
          const obj = { label: program.title, value: program.id };
          this.setState((state) => {
            const programList = state.programs.push(obj);
            return programList;
          });
        }
      }));
  }

  render() {
    const {
      nextStep, prevStep, step, promo, handleMultiChange,
    } = this.props;
    const { programs } = this.state;

    const buttonForm = (
      <div className="field buttonField section">
        <div className="control">
          <button type="button" className="button is-danger" onClick={prevStep}>Revenir</button>
        </div>
        <div className="control">
          <button type="button" className="button is-link" onClick={nextStep}>Continuer</button>
        </div>
      </div>
    );
    return (
      <div className="promoCreateForm">
        <article className="section box">
          <h1 className="title is-4 is-spaced">Cr√©ation d&apos;une promo</h1>
          <BulmaSteps step={step} />
          <div className="control">
            <span className="label">Choisir un programme : </span>
            <section className="field">
              <section className="control">
                <CreatableSelect defaultValue={promo.program} name="program" options={programs} handleChange={(e) => handleMultiChange(e, 'program')} searchKey="title" defaultLabel="Programme" />
              </section>
            </section>
          </div>
          <div className="control">
            {buttonForm}
          </div>
        </article>
      </div>
    );
  }
}

export default PromoCreateStepTwo;
