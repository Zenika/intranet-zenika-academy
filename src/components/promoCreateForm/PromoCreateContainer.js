import React, { Component } from 'react';
import { PromoCreateStepOne } from './PromoCreateStepOne';
import { PromoCreateStepTwo } from './PromoCreateStepTwo';
import PromoCreateStepThree from './PromoCreateStepThree';

export class PromoCreateContainer extends Component {
  constructor() {
    super();
    this.state = {
      step: 1,
      isTeachers: true,
      promo: {
        number: 0,
        startDate: '',
        endDate: '',
        students: [],
        teachers: [],
        program: [],
      },
    };
  }

  /**
   * Allows to handle state change in promo form
   * @param e event from wich data are coming from
   */
  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  /**
   * Allows to navigate forward on multiform
   */
  nextStep = () => {
    const { step } = this.state;
    if (step > 2) {
      this.setState({ isTeachers: false });
    }
    this.setState({ step: step + 1 });
  }

  /**
   * Allows to navigate backward on multiform
   */
  prevStep = () => {
    const { step } = this.state;
    if (step < 5) {
      this.setState({ isTeachers: true });
    }
    if (step > 0) this.setState({ step: step - 1 });
  }

  render() {
    const { step, isTeachers } = this.state;
    const { nextStep, prevStep } = this;

    switch (step) {
      case 1:
        return (
          <PromoCreateStepOne
            nextStep={nextStep}
          />
        );
      case 2:
        return (
          <PromoCreateStepTwo
            nextStep={nextStep}
            prevStep={prevStep}
          />
        );
      case 3:
        return (
          <PromoCreateStepThree
            nextStep={nextStep}
            prevStep={prevStep}
            isTeachers={isTeachers}
          />
        );
      case 4:
        return (
          <PromoCreateStepThree
            nextStep={nextStep}
            prevStep={prevStep}
            isTeachers={isTeachers}
          />
        );
      default:
        return (
          <p>error</p>
        );
    }
  }
}

export default PromoCreateContainer;
