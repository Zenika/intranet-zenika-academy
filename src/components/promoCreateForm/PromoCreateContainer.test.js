import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { PromoCreateContainer } from './PromoCreateContainer';
import PromoCreateStepOne from './PromoCreateStepOne';
import { PromoCreateStepTwo } from './PromoCreateStepTwo';
import { PromoCreateStepThree } from './PromoCreateStepThree';
import { PromoCreateStepFour } from './PromoCreateStepFour';
import PromoCreateResume from './PromoCreateResume';


Enzyme.configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = mount(<PromoCreateContainer />);
});

describe('PromoCreateContainer tests', () => {
  it('Should exist', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper.exists()).toBe(true);
  });

  it('Should have a title', () => {
    expect(global.window.document.title).toBe('CrÃ©ation de promotion');
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have PromoCreateStepOne as a child', () => {
    const container = wrapper.find(PromoCreateStepOne);
    expect(container).toHaveLength(1);
  });

  it('Should change promo name state with handleChange() on PromoCreateStepOne pomoName input', () => {
    const container = wrapper.find(PromoCreateStepOne);
    const input = container.find('#promoName');
    const mockEvent = {
      target: {
        name: 'title',
        value: 'test',
      },
    };
    input.simulate('change', mockEvent);
    expect(wrapper.state().title).toEqual('test');
  });

  it('Should change promo city state with handleChange() on PromoCreateStepOne city input', () => {
    const container = wrapper.find(PromoCreateStepOne);
    const input = container.find('#city');
    const mockEvent = {
      target: {
        name: 'city',
        value: 'Paris',
      },
    };
    input.simulate('change', mockEvent);
    expect(wrapper.state().city).toEqual('Paris');
  });

  it('Should have PromoCreateStepTwo as a child when step state === 2', () => {
    wrapper.setState({ step: 2 });
    const container = wrapper.find(PromoCreateStepTwo);
    expect(container).toHaveLength(1);
  });

  it('Should have PromoCreateStepThree as a child when step state === 3', () => {
    wrapper.setState({ step: 3 });
    const container = wrapper.find(PromoCreateStepThree);
    expect(container).toHaveLength(1);
  });

  it('Should have PromoCreateStepFour as a child when step state === 4', () => {
    wrapper.setState({ step: 4 });
    const container = wrapper.find(PromoCreateStepFour);
    expect(container).toHaveLength(1);
  });
  it('Should have PromoCreateResume as a child when step state === 5', () => {
    wrapper.setState({ step: 5 });
    const container = wrapper.find(PromoCreateResume);
    expect(container).toHaveLength(1);
  });
});
