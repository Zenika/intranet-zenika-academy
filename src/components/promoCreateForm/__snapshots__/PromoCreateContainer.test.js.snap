// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoCreateContainer tests Should match the snapshot 1`] = `
<PromoCreateContainer>
  <PromoCreateStepOne
    handleChange={[Function]}
    handleMultiChange={[Function]}
    nextStep={[Function]}
    promo={
      Object {
        "city": "",
        "country": "",
        "endDate": "",
        "program": Array [],
        "startDate": "",
        "students": Array [],
        "teachers": Array [],
        "title": "",
      }
    }
    step={1}
  >
    <div
      className="promoCreateForm"
    >
      <article
        className="section box"
      >
        <h1
          className="title is-4 is-spaced"
        >
          Création d'une promo
        </h1>
        <BulmaSteps
          step={1}
        >
          <ul
            className="steps section has-content-centered"
          >
            <li
              className="steps-segment is-active"
            >
              <span
                className="steps-marker"
              >
                <p
                  className="is-size-4"
                >
                  1
                </p>
              </span>
              <div
                className="steps-content"
              />
            </li>
            <li
              className="steps-segment  "
            >
              <span
                className="steps-marker"
              >
                <p
                  className="is-size-4"
                >
                   
                </p>
              </span>
              <div
                className="steps-content"
              />
            </li>
            <li
              className="steps-segment  "
            >
              <span
                className="steps-marker"
              >
                <p
                  className="is-size-4"
                >
                   
                </p>
              </span>
              <div
                className="steps-content"
              />
            </li>
            <li
              className="steps-segment  "
            >
              <span
                className="steps-marker"
              >
                <p
                  className="is-size-4"
                >
                   
                </p>
              </span>
              <div
                className="steps-content"
              />
            </li>
            <li
              className="steps-segment  "
            >
              <span
                className="steps-marker"
              >
                <p
                  className="is-size-4"
                >
                   
                </p>
              </span>
              <div
                className="steps-content"
              />
            </li>
          </ul>
        </BulmaSteps>
        <section
          className="control"
        >
          <label
            className="label"
            htmlFor="promoName"
          >
            Nom de la promo:
            <input
              className="input "
              defaultValue=""
              id="promoName"
              name="title"
              onChange={[Function]}
              placeholder="Promo"
              type="text"
            />
          </label>
        </section>
        <section>
          <section
            className="control"
          >
            <span
              className="label"
            >
              Date de début: 
            </span>
            <DatePicker
              date=""
              defaultValue=""
              handleChange={[Function]}
              name="startDate"
            >
              <label
                htmlFor="startDate"
              >
                <input
                  defaultValue=""
                  id="startDate"
                  name="startDate"
                  onChange={[Function]}
                  type="date"
                />
              </label>
            </DatePicker>
          </section>
        </section>
        <section>
          <section
            className="control"
          >
            <span
              className="label"
            >
              Date de fin: 
            </span>
            <DatePicker
              date=""
              defaultValue=""
              handleChange={[Function]}
              name="endDate"
            >
              <label
                htmlFor="endDate"
              >
                <input
                  defaultValue=""
                  id="endDate"
                  name="endDate"
                  onChange={[Function]}
                  type="date"
                />
              </label>
            </DatePicker>
          </section>
        </section>
        <section
          className="field"
        >
          <label
            className="label"
            htmlFor="city"
          >
            Ville:
            <section
              className="field"
            >
              <input
                className="input"
                defaultValue=""
                id="city"
                name="city"
                onChange={[Function]}
                placeholder="Ville"
              />
            </section>
          </label>
        </section>
        <section
          className="field buttonField section"
        >
          <section
            className="control"
          >
            <button
              className="button is-danger"
              id="cancelButton"
              type="button"
            >
              Annuler
            </button>
          </section>
          <section
            className="control"
          >
            <button
              className="button is-link"
              id="resumeButton"
              onClick={[Function]}
              type="button"
            >
              Continuer
            </button>
          </section>
        </section>
      </article>
    </div>
  </PromoCreateStepOne>
</PromoCreateContainer>
`;
